### PR TITLE
ExporterDirector transition step

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

Implement ExporterDirectorPartitionTransitionStep. The step re-installs exporter director when transition between Leader <-> Follower. It does nothing when transitioning between Follower <-> Candidate.

## Related issues

Related #7515 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: broker/src/test/java/io/camunda/zeebe/broker/system/partitions/TestPartitionTransitionContext.java [hunk 1]
```diff
@@ -96,6 +98,30 @@ public boolean shouldProcess() {
...
+  }
+
+  @Override
+  public PartitionMessagingService getMessagingService() {
+    return null;
+  }
+
+  @Override
+  public boolean shouldExport() {
+    return false;
```
Comment (pihme) [inline broker/src/test/java/io/camunda/zeebe/broker/system/partitions/TestPartitionTransitionContext.java:122]: Didn't we have this also in the other review?

File: broker/src/test/java/io/camunda/zeebe/broker/system/partitions/impl/steps/ExporterDirectorPartitionTransitionStepTest.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        .thenReturn(TestActorFuture.completedFuture(null));
+
+    step = new ExporterDirectorPartitionTransitionStep();
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideTransitionsThatShouldCloseExistingExporter")
+  void shouldCloseExistingStreamProcessor(final Role currentRole, final Role targetRole) {
+    // given
+    transitionContext.setCurrentRole(currentRole);
```
Comment (pihme) [inline broker/src/test/java/io/camunda/zeebe/broker/system/partitions/impl/steps/ExporterDirectorPartitionTransitionStepTest.java:65]: :thought_balloon: These four lines are repreated a couple of times. Could be refactored into a `initializeTransitionContext(...)` method. On the other hand, I like that I can read immediately what the setup does.

**Review by pihme — APPROVED** [review]
lgtm :tada: 

Some suggestions to consider

File: broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/steps/ExporterDirectorPartitionTransitionStep.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  }
+
+  private ActorFuture<Void> openExporter(
+      final PartitionTransitionContext context, final Role targetRole) {
+    final var exporterDescriptors = context.getExporterRepository().getExporters().values();
+
+    final ExporterMode exporterMode =
+        targetRole == Role.LEADER ? ExporterMode.ACTIVE : ExporterMode.PASSIVE;
+    final ExporterDirectorContext exporterCtx =
+        new ExporterDirectorContext()
```
[the remark this answers is not in the document]
Comment (deepthidevaki) [reply broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/steps/ExporterDirectorPartitionTransitionStep.java:86]: I would keep it as it is. I don't want to make `PartitionTransitionContext` aware of how to build an exporter.

**Pull request merged**